src/main/java/com/minecolonies/items/ItemSupplyChestDeployer.java

```diff
@@ -173,83 +174,119 @@ private void spawnShip(@NotNull World world, @NotNull BlockPos pos, EntityPlayer
      * @param isCoordPositivelyAdded boolean whether the x or z side should be check on the positive side (true) or negative  side (false).
      * @return whether the space in the I shape is free or not.
      */
-    private boolean check(@NotNull World world, @NotNull BlockPos pos, boolean shouldCheckX, boolean isCoordPositivelyAdded)
+    private static boolean check(@NotNull final World world, @NotNull final BlockPos pos, final boolean shouldCheckX, final boolean isCoordPositivelyAdded)
     {
-        int k = isCoordPositivelyAdded ? 1 : -1;
+        final int k = isCoordPositivelyAdded ? 1 : -1;
 
+        final int spaceRightK = SPACE_RIGHT * k;
+        final int spaceLeftK = SPACE_LEFT * k;
+        final int widthK = WIDTH * k;
 
-        int horizontalX = isCoordPositivelyAdded ? SPACE_LEFT : SPACE_RIGHT;
-        int horizontalZ = isCoordPositivelyAdded ? SPACE_RIGHT : SPACE_LEFT;
-
-        int spaceRightK = SPACE_RIGHT * k;
-        int spaceLeftK = SPACE_LEFT * k;
+        if (shouldCheckX)
+        {
+            return checkX(world, pos, k, spaceRightK, spaceLeftK, widthK, isCoordPositivelyAdded);
+        }
 
-        int widthK = WIDTH * k;
-        int widthKHalf = widthK / 2;
+        return checkZ(world, pos, k, spaceRightK, spaceLeftK, widthK, isCoordPositivelyAdded);
+    }
 
-        if (shouldCheckX)
+    /**
+     * Checks the z axis for water.
+     * @param world the world.
+     * @param pos the starting pos.
+     * @param k the symbol.
+     * @param spaceRightK the space to the left.
+     * @param spaceLeftK the space to the right.
+     * @param widthK the width.
+     * @param isCoordPositivelyAdded if is positive or not.
+     * @return true if it can be placed.
+     */
+    private static boolean checkZ(final World world, final BlockPos pos, final int k, final int spaceRightK, final int spaceLeftK, final int widthK, final boolean isCoordPositivelyAdded)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Split this 186 characters long line (which is greater than 180 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00103)